### PR TITLE
Add GitHub templates for issues and PRs

### PR DESCRIPTION
**Pull request by tammullen**
## Changes proposed in this Pull Request:

This PR adds GitHub issue forms templates, issue chooser, and a pull request template (and associated contribution guidelines) to the repository.

**Issue forms template files added:**
`1-bug-report.yml` - Bug Report template using issue forms
`2-enhancement.yml` - Enhancement template using issue forms
`config.yml` - Issue Chooser (what will be shown when clicking `/issues/new/choose`) config and additional links  
`ISSUE_TEMPLATE.md` - Template for "Blank" issues

**Pull Request template files added:**
`PULL_REQUEST_TEMPLATE.md` - Template for Pull Requests
`CONTRIBUTING.md` - Basic contributing guidelines referenced in the Pull Request template

## More details:

The proposed issue chooser will display all the `.yml` templates in `.github/ISSUE_TEMPLATE` and pulls the description for each from the template itself.
The order of the templates is determined by the alphabetical order of the file names of the templates.

The proposed additional `config.yml` file will control the content below the templates of the issue chooser and allows us to include external links. (These links can only be displayed after the templates).

More details on configuring `conifig.yml` here https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#configuring-the-template-chooser

## How to test the changes in this Pull Request:

- I don't know of a way to visually preview the issue chooser and the descriptions external links in `config.yml`
- To preview the issue templates view the file in the Files changed tab and GitHub will render a visual preview. (The section at the top is for the description that will be on the issue template chooser so it can be ignored).

### CODE REVIEW

File: .github/ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<!-- This form is for other issue types specific to WooCommerce E2E Tests Boilerplate. This is not a support portal. -->
```
Comment (zhongruige) [inline .github/ISSUE_TEMPLATE.md]: Minor change here to that this for Smooth Generator not for WooCommerce E2E Tests Boilerplate:

```suggestion
<!-- This form is for other issue types specific to WooCommerce Smooth Generator. This is not a support portal. -->
```

File: .github/ISSUE_TEMPLATE/2-enhancement.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+name: ✨ Enhancement Request
+description: If you have an idea to improve WooCommerce E2E Tests Boilerplate or need something for development please let us know or submit a Pull Request!
```
Comment (zhongruige) [inline .github/ISSUE_TEMPLATE/2-enhancement.yml]: We should swap this to be WooCommerce Smooth Generator:

```suggestion
description: If you have an idea to improve WooCommerce Smooth Generator or need something for development please let us know or submit a Pull Request!
```

**Review by zhongruige — CHANGES_REQUESTED** [review]
Thanks for adding this in @tammullen!

Overall looks good, just a few minor tweaks are needed before we can merge this in.

**Comment by rrennick** [comment]
@ObliviousHarmony Can your team review this PR as well?

**Comment by ObliviousHarmony** [comment]
@rrennick @tammullen Just going to echo the same feedback as @zhongruige already left on the pull request! Everything else looks good!

**Comment by tammullen** [comment]
Corrected typos, thanks for catching them @zhongruige !

**Review by zhongruige — APPROVED** [review]
Thanks for taking care of those @tammullen!

LGTM :shipit:

**Pull request merged**